packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,646 @@
+attribute to acquire the regular expression pattern that corresponds to the URLs of users within the specific domain of the client application.
+
+The `nimble-mapping-mention-user` element contains individual user information within its elements. Typically, the `url` attribute
+includes either the user ID or a value stored in the backend database. It's important that the `url` adheres to valid URL format,
+often used to link the user ID with the hosting domain. In cases where there's no specific URL for each user within the domain,
+they can alternatively insert the user ID in a valid URL format such as `user:user-id` and provide the appropriate pattern for it.
+
+#### Future Scope
+
+1.  The `nimble-mapping-mention-user` can be used to get other user details like email, profile information etc., to display in the
+    dropdown list options.
+2.  If there is a requirement to mention an issue (using **"#"**) or a pull requests (using **"!"**),
+    a new configuration component can be created and added as a child to the rich text editor. Below is an example of how
+    these elements would be used:
+
+    ````html
+    <nimble-rich-text-editor>
+        <nimble-rich-text-mention-users pattern="http://users/.*">
+            <nimble-mapping-mention-user
+                url="http://users/user-id-1"
+                display-name="John Doe"
+            ></nimble-mapping-mention-user>
+            <nimble-mapping-mention-user
+                url="http://users/user-id-1"
+                display-name="Alice Smith"
+            ></nimble-mapping-mention-user>
+            <nimble-mapping-mention-user
+                url="http://users/user-id-1"
+                display-name="Bob Jones"
+            ></nimble-mapping-mention-user>
+        </nimble-rich-text-mention-users>
+
+        <nimble-rich-text-mention-issues pattern="http://issues/.*">
+            <nimble-mapping-mention-user
+                url="http://issue/issue-id-1"
+                display-name="Spec for rich text editor"
+            ></nimble-mapping-mention-user>
+            <nimble-mapping-mention-user
+                url="http://issue/issue-id-2"
+                display-name="Mention support in rich text components"
+            ></nimble-mapping-mention-user>
+            <nimble-mapping-mention-user
+                url="http://issue/issue-id-3"
+                display-name="Issue in pasting a link"
+            ></nimble-mapping-mention-user>
+        </nimble-rich-text-mention-issues>
+    </nimble-rich-text-editor>
+    ```
+    ````
+
+3.  As the URLs are stored as absolute links within the markdown string, mentions of issues or systems, as described above,
+    initially appear as links. However, once we introduce support for mentioning issues or systems by matching the specified
+    pattern in the configuration element, even the absolute links will transform into `mention` nodes.
```

Comment:
What about virtualizing the set of mention users, where we may need to handle in the thousands of user mentions, but don't want a DOM presence for all of them?